### ISSUE TITLE
Using on server side

### ISSUE THREAD

**Issue by Sunil6591** [davidkpiano/xstate#638]
Here is state machine config
```js
export const requestMachine = Machine({
  id: 'request',
  initial: 'draft',
  context: {
    retries: 0,
    comments: [],
  },
  states: {
    draft: {
      on: {
        SAVE: {
          target: 'draft',
          actions: ['saveItToDB'],
        },
        SUBMIT: {
          target: 'pending',
          actions: ['saveItToDB'],
        },
      },
    },
    pending: {
      on: {
        APPROVE: {
          target: 'approve',
          actions: ['approveIt'],
        },
        REJECT: 'reject',
        ON_HOLD: 'hold',
        SAVE: {
          target: 'pending',
        },
      },
    },
    hold: {
      on: {
        APPROVE: {
          target: 'approve',
          actions: ['approveIt'],
        },
        REJECT: 'reject',
      },
    },
    approve: {
      type: 'final',
    },
    reject: {
      type: 'final',
    },
  },
},
{
  actions: {
    saveItToDB: (context, event) => {
      console.log('saveItToDB ***************', context.request, event)
    },
    approveIt: (context, event) => {
      console.log('Approve it ***************', Object.keys(context), event)
    },
  },
}
);
```

Now in the `actions` e.g. `saveItToDB` I get only two params `context` and `event`.
How do I get current state of machine so I can save it in DB along with the `request` object. 
The reason I want to save the current state, is that next time when user on frontend invokes `APPROVE` event, i can pull the current state of machine and then start the intercepter service with that state. So that the machine continues and also I have the history of events

In short the question is - In actions is there a way to get current state of machine?

https://xstate.js.org/viz/?gist=ef97fc9788343f0def46b03711dc696e

**Comment by Andarist** [comment]
Actions get also third argument - meta, `state` should be available on that object.

**Comment by davidkpiano** [comment]
Yep, you would do it like this:

```js
    saveItToDB: (context, event, { state }) => {
      console.log(state); // here's your state!
      console.log('saveItToDB ***************', context.request, event)
    },
```

I think this is in the docs, I'll check.

**Comment by Sunil6591** [comment]
Thanks a ton! Simply ❤️ xstate
